frontend/tests/e2e-tests.sh

```
@@ -200,9 +226,16 @@
-	echo "| quick mode"
-	npx playwright test ./tests/functional/"${TEST_PATHS[@]}" -x --project=chromium "${SCRIPT_LONG_ARGS[@]}"
+FRONTEND_HASH_FILE="$APP_DIR/frontend/tests/.frontend_hash"
+FRONTEND_HASH=$(find "$APP_DIR/frontend/src" -type f \( -name "*.ts" -o -name "*.svelte" \) | xargs -I {} md5sum {} | md5sum)
+
+if ! cmp <(cat "$FRONTEND_HASH_FILE") <(echo "$FRONTEND_HASH") &>/dev/null; then
+	pnpm run build # Required for the "pnpm run preview" command of playwright.config.ts
+	echo "$FRONTEND_HASH" > "$FRONTEND_HASH_FILE"
+fi
+
+if [[ -n "$QUICK_MODE_ACTIVATED" ]]; then
+	pnpm playwright test ./tests/functional/"${TEST_PATHS[@]}" --project=chromium "${SCRIPT_LONG_ARGS[@]}" "${SCRIPT_SHORT_ARGS[@]}"
```

Shellcheck (0.10.0)

[error] Argument mixes string and array. Use * or separate argument.

(SC2145)